scripts/src/java/org/oppia/android/scripts/common/MavenDependenciesListGenerator.kt

```diff
@@ -0,0 +1,247 @@
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<License> {
+    return mavenDependenciesList.flatMap { dependency ->
+      dependency.licenseList.filter { license ->
+        license.verifiedLinkCase != License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET
+      }
+    }.toSet()
+  }
+
+  fun updateMavenDependenciesList(
+    latestDependenciesList: List<MavenDependency>,
+    manuallyUpdatedLicenses: Set<License>
+  ): List<MavenDependency> {
+    return latestDependenciesList.map { mavenDependency ->
+      val updatedLicenseList = mavenDependency.licenseList.map { license ->
+        manuallyUpdatedLicenses.find {
+          it.originalLink == license.originalLink && it.licenseName == license.licenseName
+        } ?: license
+      }
+      MavenDependency.newBuilder().apply {
+        this.artifactName = mavenDependency.artifactName
+        this.artifactVersion = mavenDependency.artifactVersion
+        this.addAllLicense(updatedLicenseList)
+      }.build()
+    }
+  }
+
+  fun writeTextProto(
+    pathToTextProto: String,
+    mavenDependencyList: MavenDependencyList
+  ) {
+    File(pathToTextProto).outputStream().bufferedWriter().use { writer ->
+      TextFormat.printer().print(mavenDependencyList, writer)
+    }
+  }
+
+  fun getAllBrokenLicenses(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<License> {
+    // Here broken licenses are those licenses that do not have verified_link set or
+    // there original link is found to be invalid by the developers.
+    return mavenDependenciesList.flatMap { dependency ->
+      dependency.licenseList.filter { license ->
+        license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET) &&
+          !license.isOriginalLinkInvalid
+      }
+    }.toSet()
+  }
+
+  fun findFirstDependenciesWithBrokenLicenses(
+    mavenDependenciesList: List<MavenDependency>,
+    brokenLicenses: Set<License>
+  ): Map<License, String> {
+    return brokenLicenses.associateTo(mutableMapOf<License, String>()) { license ->
+      license to mavenDependenciesList.first { dependency ->
+        dependency.licenseList.contains(license)
+      }.artifactName
+    }
+  }
+
+  fun getDependenciesThatNeedIntervention(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<MavenDependency> {
+    // The dependencies whose license list is empty or some of their license link was found to
+    // be invalid need intervention. In this case, the developer needs to manually fill in
+    // the license links for each of these dependencies.
+    return mavenDependenciesList.filter { dependency ->
+      dependency.licenseList.isEmpty() ||
+        dependency.licenseList.filter { license ->
+          license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET) &&
+            license.isOriginalLinkInvalid
+        }.isNotEmpty()
+    }.toSet()
+  }
+
+  /** Retrieves the list of [MavenDependency] from maven_dependencies.textproto. */
```

Comment:
`@param` missing

Reply:
Done.

Reply:
doesn't seems like done. 

What I meant here is we can put the detail, same as what we do while adding general kdoc for any function. 

```kt
@param [pathToPbFile]: ...

@return ...
```

Reply:
Done.